internal/services/machinelearning/machine_learning_datastore_blobstorage_resource.go

```diff
@@ -0,0 +1,268 @@
+	if err != nil {
+		return fmt.Errorf("parsing Machine Learning Data Store ID `%q`: %+v", d.Id(), err)
+	}
+
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on Machine Learning Data Store %q (Resource Group %q): %+v", id.Name, id.ResourceGroupName, err)
+	}
+
+	workspaceId := workspaces.NewWorkspaceID(subscriptionId, id.ResourceGroupName, id.WorkspaceName)
+	d.Set("name", resp.Model.Name)
+	d.Set("workspace_id", workspaceId.ID())
+	return flattenBlobStorage(d, resp.Model.Properties.(datastore.AzureBlobDatastore))
+}
+
+func resourceMachineLearningDataStoreDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MachineLearning.DatastoreClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := datastore.ParseDataStoreID(d.Id())
+	if err != nil {
+		return fmt.Errorf("parsing Machine Learning Workspace Date Store ID `%q`: %+v", d.Id(), err)
+	}
+
+	if _, err := client.Delete(ctx, *id); err != nil {
+		return fmt.Errorf("deleting Machine Learning Workspace Date Strore %q (Resource Group %q): %+v", id.Name, id.ResourceGroupName, err)
+	}
+
+	return nil
+}
+
+func expandBlobStorage(d *pluginsdk.ResourceData) *datastore.AzureBlobDatastore {
+	storeProps := &datastore.AzureBlobDatastore{
+		AccountName:                   utils.String(d.Get("storage_account_name").(string)),
+		ContainerName:                 utils.String(d.Get("container_name").(string)),
+		Description:                   utils.String(d.Get("description").(string)),
+		ServiceDataAccessAuthIdentity: utils.ToPtr(datastore.ServiceDataAccessAuthIdentity(d.Get("service_data_auth_identity").(string))),
+		IsDefault:                     utils.Bool(d.Get("is_default").(bool)),
+		Tags:                          utils.ToPtr(expandTags(d.Get("tags").(map[string]interface{}))),
+	}
+
+	accountKey := d.Get("account_key").(string)
+	if accountKey != "" {
+		storeProps.Credentials = map[string]interface{}{
+			"credentialsType": string(datastore.CredentialsTypeAccountKey),
+			"secrets": map[string]interface{}{
+				"secretsType": "AccountKey",
+				"key":         accountKey,
+			},
+		}
+	}
+
+	sasToken := d.Get("shared_access_signature").(string)
+	if sasToken != "" {
+		storeProps.Credentials = map[string]interface{}{
+			"credentialsType": string(datastore.CredentialsTypeSas),
+			"secrets": map[string]interface{}{
+				"secretsType": "Sas",
+				"sasToken":    sasToken,
+			},
+		}
+	}
+
+	return storeProps
+}
+
+func flattenBlobStorage(d *pluginsdk.ResourceData, data datastore.AzureBlobDatastore) error {
+	d.Set("description", data.Description)
+	d.Set("is_default", data.IsDefault)
+	d.Set("service_data_auth_identity", string(*data.ServiceDataAccessAuthIdentity))
+	d.Set("storage_account_name", *data.AccountName)
+	d.Set("container_name", *data.ContainerName)
```

Comment:
These should be nil checked and always have a value set into state:
```suggestion
	serviceDataAuth := ""
	if v := data.ServiceDataAccessAuthIdentity; v != nil {
		serviceDataAuth = string(*v)
	}
	d.Set("service_data_auth_identity", serviceDataAuth)
	
	storageAccountName := ""
	if v := data.AccountName; v != nil {
		storageAccountName = *v
	}
	d.Set("storage_account_name", storageAccountName)
	
	containerName := ""
	if v := data.ContainerName; v != nil {
		containerName = *v
	}	
	d.Set("container_name", containerName)
```